lib/actions/api.js

```diff
@@ -775,6 +765,54 @@ function createQueryAction (endpoint, responseAction, errorAction, options = {})
   }
 }
 
+/**
+ * Creates the URL to use for making an API request.
+ *
+ * @param  {Object} config   The app-wide config
+ * @param  {string} endpoint The API endpoint path
+ * @param  {Object} options  The options object for the API request
+ * @return {string}          The URL to use for making the http request
+ */
+function makeApiUrl (config, endpoint, options) {
+  let url
+  if (
+    options.serviceId &&
+      config.alternateTransitIndex &&
+      config.alternateTransitIndex.services.includes(options.serviceId)
+  ) {
+    console.log('Using alt service for ' + options.serviceId)
+    url = config.alternateTransitIndex.apiRoot + endpoint
+  } else {
+    const api = config.api
+    url = `${api.host}${api.port ? ':' + api.port : ''}${api.path}/${endpoint}`
+  }
+  return url
+}
+
+/**
+ * Add the necessary API key headers if using an API Key.
+ *
+ * @param  {Object} config  The app-wide config
+ * @param  {Object} options The options for the API request
+ * @return {Object}         The options to send to the fetch function when
+ *                      making the http request for the API endpoint.
+ */
+function makeApiFetchOptions (config, options) {
+  const fetchOptions = options.fetchOptions || {}
+  const apiKey = config?.persistence?.otp_middleware?.apiKey
+  if (apiKey) {
+    const headers = fetchOptions.headers || {}
+    fetchOptions.headers = {
+      'x-api-key': apiKey,
+      ...headers
+    }
+    if (!fetchOptions.mode) {
+      fetchOptions.mode = 'cors'
+    }
```

Comment:
There's a bit of shared code here with `getSecureFetchOptions`. Did you consider any reuse? Given that they inject a lot of the same headers, I think that might be worthwhile.

Reply:
I did consider reuse, but only the apiKey and cors thing are needed, so I figured it wasn't worth the effort to add extra configuration arguments to that method.

Reply:
See #445 for some proposed changes.